front/lib/actions/retrieval.ts

```diff
@@ -368,6 +369,20 @@ export class RetrievalConfigurationServerRunner extends BaseActionConfigurationS
     RetrievalParamsEvent | RetrievalSuccessEvent | RetrievalErrorEvent,
     void
   > {
+    const killSwitches = await KillSwitchResource.listEnabledKillSwitches();
```

Comment:
LGTM

We'll want to revert as it adds a DB load in our critical path (ideally we parallelize the load higher up with something else).

Reply:
👍